static/js/disaster_dashboard/map_section.tsx

```diff
@@ -0,0 +1,252 @@
+
+const DISASTER_COLORS = {
+  [DisasterType.EARTHQUAKE]: "red",
+  [DisasterType.FIRE]: "orange",
+  [DisasterType.STORM]: "yellow",
+  [DisasterType.FLOOD]: "green",
+  [DisasterType.DROUGHT]: "blue",
+};
+
+interface MapSectionPropType {
+  disasterEventPoints: DisasterEventPoint[];
+  geoJson: GeoJsonData;
+  selectedDisaster: DisasterType;
+  selectedDate: string;
+  selectedPlaceInfo: PlaceInfo;
+  selectedIntensityProp: string;
+  onPlaceUpdated: (place: NamedPlace) => void;
+  europeanPlaces: NamedPlace[];
+}
+
+export function MapSection(props: MapSectionPropType): JSX.Element {
+  const svgContainer = useRef(null);
+  const infoCardRef = useRef(null);
+
+  useEffect(() => {
+    if (
+      _.isEmpty(props.geoJson) ||
+      props.geoJson.properties.current_geo !==
+        props.selectedPlaceInfo.place.dcid
+    ) {
+      return;
+    }
+    draw();
+    removeSpinner(CONTENT_SPINNER_ID);
+  }, [props.disasterEventPoints, props.geoJson]);
+
+  if (_.isEmpty(props.geoJson)) {
+    removeSpinner(CONTENT_SPINNER_ID);
+    return (
+      <div className="disaster-dashboard-map-section">
+        <div className="error-message">
+          Sorry, we do not have maps for this place.
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="disaster-dashboard-map-section">
+      <div className="zoom-button-section">
+        <div id={ZOOM_IN_BUTTON_ID} className="zoom-button">
+          <i className="material-icons">add</i>
+        </div>
+        <div id={ZOOM_OUT_BUTTON_ID} className="zoom-button">
+          <i className="material-icons">remove</i>
+        </div>
+      </div>
+      <div id={MAP_ID} ref={svgContainer}></div>
+      {props.selectedDisaster === DisasterType.ALL ? (
+        <div className="all-disasters-legend">
+          {Object.keys(DISASTER_COLORS).map((disasterType) => {
+            return (
+              <div className="all-disasters-legend-entry" key={disasterType}>
+                <div
+                  className="all-disasters-legend-color"
+                  style={{ backgroundColor: DISASTER_COLORS[disasterType] }}
+                ></div>
+                <span>{disasterType}</span>
+              </div>
+            );
+          })}
+        </div>
+      ) : (
+        <div id={LEGEND_CONTAINER_ID}></div>
+      )}
+      <div id="disaster-dashboard-info-card" ref={infoCardRef} />
+    </div>
+  );
+
+  function onPointClicked(
+    point: DisasterEventPoint,
+    event: PointerEvent
+  ): void {
+    const infoCard = d3.select(infoCardRef.current);
+    ReactDOM.render(
+      <InfoCard
+        onClose={() =>
+          d3.select(infoCardRef.current).style("visibility", "hidden")
+        }
+        eventData={point}
+      />,
+      infoCardRef.current
+    );
+    const infoCardHeight = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect().height;
+    const infoCardWidth = (
+      infoCard.node() as HTMLDivElement
+    ).getBoundingClientRect().width;
```

Comment:
can we cache the results of getBoundingClientRect (which is somewhat expensive), and then read the values of height and width?

Reply:
please also add some high-level comments, e.g. compute card dimensions, etc

Reply:
done